### ISSUE TITLE
Compilation errors ensue when using `--parallel`.

### ISSUE THREAD

**Issue by noel-yap** [johnrengelman/shadow#101]
The behavior of the plugin is very different when using versus not using `--parallel` (in mult-module builds, of course).

**Comment by TomaszDudziak** [comment]
I also experience the plugin failing in parallel mode for multi-module build.
`gradle --parallel clean test` works
`gradle clean test shadowJar` works
`gradle --parallel clean test shadowJar` fails with:
```
Execution failed for task ':myProject:myModule:test'.
> failed to scan jar file for packages (C:\sandbox\myProject\myModule\build\libs\myModule.jar)
```
Note the plugin does build shadow jars for some of the modules, but then fails for one.

**Comment by johnrengelman** [comment]
Can you try running with `-i` or -`d` to see what's happening? Shadow doesn't interact with the `test` task, so from that error message it doesn't seem to be a problem with Shadow at all.
Also, try doing `gradle clean && gradle --parallel test shadowJar` and see if you get the same failure.

**Comment by TomaszDudziak** [comment]
Thanks for a quick reply. 
`gradle clean`
followed by
`gradle --parallel test shadowJar`
resulted in 
```
Execution failed for task ':myProject:myOtherModule:compileScala'.
> Compile failed because of an internal compiler error (object java.lang.Object in compiler mirror not found.); see the error output for details.
```
I can post the full stack trace if you like.

**Comment by johnrengelman** [comment]
That doesn't appear to have anything to do with Shadow.

**Comment by TomaszDudziak** [comment]
I see... well maybe it could be a problem with gradle then. All I know is that:
`gradle clean test shadowJar` consistently works and
`gradle --parallel clean test shadowJar` (or `gradle clean` followed by `gradle --parallel test shadowJar`) consistently does not work

**Comment by johnrengelman** [comment]
Do you get failures doing `gradle --parallel clean test`?

**Comment by TomaszDudziak** [comment]
OK, I have found a workaround. Looks like this is a problem with gradle.
So, summing up:
`gradle clean test shadowJar` works
`gradle --parallel clean test shadowJar` does not work
`gradle --parallel clean test` works
`gradle --parallel test shadowJar` does not work
but:
`gradle --parallel shadowJar` works!
so I can run:
`gradle --daemon --parallel clean test`
and then:
`gradle --daemon --parallel shadowJar`
Looks like this could be a problem with gradle running certain tasks together in the parallel mode. Anyway, we've got a workaround.
